src/app/util/ember-compatibility-functions.cpp

```diff
@@ -860,29 +860,32 @@ CHIP_ERROR WriteSingleClusterData(const SubjectDescriptor & aSubjectDescriptor,
         return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::NeedsTimedInteraction);
     }
 
-    if (auto * attrOverride = findAttributeAccessOverride(aClusterInfo.mEndpointId, aClusterInfo.mClusterId))
+    uint16_t dataLen            = 0;
+    CHIP_ERROR preparationError = prepareWriteData(attributeMetadata, aReader, dataLen);
```

Comment:
So this is going to duplicate some work for attributes of types that _could_ be handled by the attribute store but have overrides registered, right?  All we're trying to determine here is whether the `size` of the `attributeMetadata` is meaningful, right?  And all we're _really_ trying to determine is whether strings fit in the size they're supposed to fit in.  Can we just do the check here for string/octstr types and not move around where `prepareWriteData` happens?